src/js/actions/debug.js

```diff
@@ -374,43 +400,124 @@ define(function (require, exports) {
-            log.debug("[Descriptor] PS Event: %c%s\n%c%s", eventStyle, eventID, blockStyle,
+            log.debug("[Descriptor] Received PS Event: %c%s\n%c%s", "font-weight:bold", eventID, blockStyle,
                 JSON.stringify(obj, null, " "));
         };
 
         if (enabled) {
+            // Customized version of the `descriptor._getAsync` to capture the descriptor ID.
+            var getAsync = Promise.promisify(function (reference, options, idCallback, callback) {
+                var id = this.get(reference, options, callback);
+                idCallback(id);
+                return id;
+            }, {
+                context: window._spaces.ps.descriptor
+            });
+            
             descriptor._getAsync = function (reference, options) {
-                log.debug("[Descriptor] get: \n%c%s", blockStyle,
-                    JSON.stringify(reference, null, ""));
+                var start = new Date(),
+                    descriptorID;
+
+                return getAsync(reference, options, function (id) {
+                    descriptorID = id;
+
+                    log.groupCollapsed("%c[Descriptor] Executing get - %d", groupStyle, descriptorID);
+                    log.trace("Trace");
+                    log.debug("Params:\n%c%s", blockStyle, JSON.stringify(reference, null, " "));
+                    log.groupEnd();
+                    log.timeStamp("[Descriptor] Executing get - " + descriptorID);
+                }).then(function (result) {
+                    var end = new Date();
+
+                    log.groupCollapsed("%c[Descriptor] Finished get - %d in %c%dms", groupStyle,
+                        descriptorID,
+                        end - start > SLOW_DESCRIPTOR ? redText : blueText,
+                        end - start);
+                    log.debug("Result:\n%c%s", blockStyle, JSON.stringify(result, null, " "));
+                    log.groupEnd();
+                    log.timeStamp("[Descriptor] Finished get - " + descriptorID);
+                    return result;
+                });
+            };
 
-                return descriptor.__getAsync(reference, options).then(function (result) {
-                    log.debug("[Descriptor] get result: \n%c%s", blockStyle,
-                        JSON.stringify(result, null, "  "));
+            // Customized version of the `descriptor._batchPlay` to capture the descriptor ID.
+            var batchPlayAsync = Promise.promisify(function (commands, options, idCallback, callback) {
+                var id = this.batchPlay(commands, options, callback);
+                idCallback(id);
+                return id;
+            }, {
+                context: window._spaces.ps.descriptor,
+                multiArgs: true
+            });
 
-                    return result;
+            var summarizeCommands = function (commands) {
+                if (commands.length === 0) {
+                    return "EMPTY";
```

Comment:
Can we just make this the empty string here so this shows up as `Executing batchPlay: [] - 123`?